llvm/lib/Bitcode/Writer/BitcodeWriter.cpp

```diff
@@ -99,6 +99,21 @@ static cl::opt<bool> WriteRelBFToSummary(
     "write-relbf-to-summary", cl::Hidden, cl::init(false),
     cl::desc("Write relative block frequency to function summary "));
 
+// Since we only use the context information in the memprof summary records in
+// the LTO backends to do assertion checking, save time and space by only
+// serializing the context for non-NDEBUG builds.
+// TODO: Currently this controls writing context of the allocation info records,
+// which are larger and more expensive, but we should do this for the callsite
+// records as well.
+static cl::opt<bool>
+    CombinedIndexMemProfContext("combined-index-memprof-context", cl::Hidden,
+#ifndef NDEBUG
```

Comment:
What is the value of a flag instead of a constant? 

I think in a NDEBUG build then the assertions do not exist downstream. So why would need to change the behaviour to send the context information, are there other uses apart from assertions?

Reply:
Added a FIXME to convert to a const after more testing. I'd like a way to disable this change if it hits unexpected issues.

Reply:
Sounds good to me.